src/reactapp/src/pages/EvaluationForm.js

```diff
@@ -4,74 +4,181 @@
+        setPerformanceEvaluationData(retrievedPerformanceEvaluationData);
+        setEvaluationDate(new Date(retrievedPerformanceEvaluationData.startDateTime+"Z"));
+        setCurrentEvaluator({ "evaluatorId": retrievedPerformanceEvaluationData.userId, "evaluatorName": retrievedPerformanceEvaluationData.evaluatorName });
+        const candidateReceived = {
+          candidateName: retrievedPerformanceEvaluationData.reviewedCandidateName,
+          personId: retrievedPerformanceEvaluationData.reviewedPersonId,
+          sourceSystemDescriptor: retrievedPerformanceEvaluationData.reviewedPersonSourceSystemDescriptor
+        }
+        const evaluationHeader = {
+          id: retrievedPerformanceEvaluationData.evaluationId,
+          performanceEvaluationTitle: retrievedPerformanceEvaluationData.performanceEvaluationTitle
+        }
+        sessionStorage.setItem("evaluation", JSON.stringify(evaluationHeader));
+        setSelectedEvaluation(evaluationHeader);
+        sessionStorage.setItem("candidate", JSON.stringify(candidateReceived));
+        setSelectedCandidate(candidateReceived);
+        const startDate = new Date(retrievedPerformanceEvaluationData.startDateTime);
+        const endDate = new Date(retrievedPerformanceEvaluationData.startDateTime);
+        if (endDate > startDate) {
+          setEvaluationEndTime(new Date(endDate - startDate));
+        }        
+      }
+    } catch (error) {
+      console.error("Error fetching performance evaluation:", error);
+    }
+  };
+
+  const loadDataForNewEvaluation = () => {
+    setCurrentEvaluator({ "evaluatorId": getLoggedInUserRole(), "evaluatorName": getLoggedInUserName() });
     if (location?.state?.candidate) {
       sessionStorage.setItem("candidate", JSON.stringify(location.state.candidate));
       setSelectedCandidate(location.state.candidate);
     } else {
       setSelectedCandidate(JSON.parse(sessionStorage.getItem("candidate")));
     }
-
     if (location?.state?.evaluation) {
       sessionStorage.setItem("evaluation", JSON.stringify(location.state.evaluation));
       setSelectedEvaluation(location.state.evaluation);
     } else {
       setSelectedEvaluation(JSON.parse(sessionStorage.getItem("evaluation")));
     }
-  }, []);
+  };   
 
+  useEffect(() => {
+    setIsEvaluationLoaded(false);
+    setLoggedInUser({
+      "name": getLoggedInUserName(),
+      "role": getLoggedInUserRole()
+    });
+    
+    if (id) {
+      loadExistingEvaluation();
+    }
+    else {
+      setCurrentEvaluator({ "evaluatorId": getLoggedInUserId(), "evaluatorName": getLoggedInUserName() });
+      loadDataForNewEvaluation();
+    }
+    setIsEvaluationLoaded(true);
+  }, [id]);
+
+  
   const processRatingLevelOptions = async (ratingLevels) => {
-    const processedRatingLevels = [{ "label": "N/A - Not Assessed", "value": -1 }, ...ratingLevels.map((level) => {
+    const processedRatingLevels = [{ "label": "N/A - Not Applicable", "value": -1 }, ...ratingLevels.map((level) => {
```

Comment:
@jleiva-gap Please keep the previous terminology of 'Not Assessed'

Reply:
Done